### ISSUE TITLE
IOS version

### ISSUE THREAD

**Issue by roozbehsat** [vpnhood/VpnHood#235]
Please release the ios version...

**Comment by zhangkjun** [comment]
Please release the ios version...

**Comment by vikofun** [comment]
The iso version is really important. Without the iso version, I feel that it is half the sky less.

**Comment by francescovigot** [comment]
How can we be informed if the IOS version publish?

**Comment by trudyhood** [comment]
> How can we be informed if the IOS version publish?

You can subscribe to this issue. Unfortunately, we don't have enough resources to implement it at this moment.
![image](https://user-images.githubusercontent.com/72891124/223213099-2b161b87-b0f5-4bdc-90b1-148f61985c87.png)

**Comment by asoMan1369** [comment]
is it possible to develop the IOS version by React native ?

**Comment by trudyhood** [comment]
> is it possible to develop the IOS version by React native?

The UI is already a progressive web; I don't think the ReactNative can help. Everything is ready for iOS. We just need to implement this interface on iOS then package it via Xamarin and publish it to Apple Store:
https://github.com/vpnhood/VpnHood/blob/main/VpnHood.Client.Device/IPacketCapture.cs

Same as what we did for android:
https://github.com/vpnhood/VpnHood/blob/main/VpnHood.Client.Device.Android/AndroidPacketCapture.cs

**Comment by asoMan1369** [comment]
Ok it's Great.
I know someone who is interested to participate on project.
she is working with xamarin , yesterday I was speaking with her and she told me that is able to develop IOS by xamarin
I will connect her with you.

**Comment by trudyhood** [comment]
@asoMan1369 Did you manage to contact her?

**Comment by asoMansoury** [comment]
> 

Yes, unfortunately, she didn't accept and said she does not have enough time to spend on the project!

**Comment by roozbehsat** [comment]
Any News about ios version?

**Comment by trudyhood** [comment]
Nope. 😞

**Comment by vedanshi-systems** [comment]
I am working on it.  @trudyhood will appreciate you help in the process

**Comment by justus-go** [comment]
Any new news?

**Comment by 0xBF90E913** [comment]
any news? ahah, i realy very need ios version, is there any way to do that? With crutches, at least
@trudyhood @vedanshi-systems

**Comment by miladxandi** [comment]
When this issue resolved, I will publish one iOS application.
https://github.com/vpnhood/VpnHood/issues/542

**Comment by shap4ever** [comment]
Another year is passed. Any update on this?

**Comment by trudyhood** [comment]
> Another year is passed. Any update on this? You’ve done an incredible job with this project—it’s really impressive! That said, without an iOS app, it feels a bit incomplete. If you're open to it, consider sharing a Patreon or fundraiser link so people can support you in hiring developers to help finish it.

We have built everything necessary for iOS, and it is functional. However, the person responsible for publishing and packaging for the Apple Store was no longer able to continue, so unfortunately, the iOS version remains on hold at this time.

**Comment by shap4ever** [comment]
Thanks, Trudy. I still believe it's best to ask for help—otherwise, it might remain blocked indefinitely. I understand this is non-profit work, and no one expects you to sacrifice your personal life for it.

**Comment by topninja** [comment]
@trudyhood 
Can I help to finish ios version?

**Comment by trudyhood** [comment]
> [@trudyhood](https://github.com/trudyhood) Thanks for the overview. Just to clarify—when you say "packing," do you mean preparing the app for App Store distribution? I was under the impression that the project (VpnHood?) is open-source. If that's the case, is there any specific reason we need to "pack" it instead of just integrating or referencing the open components?
> 
> Let me know how you'd like to proceed.

By packing, I mean assembling the app: setting up the structure, linking the components, and preparing it for App Store distribution.

VpnHood is open-source, so all code will be public in the repository. However, on iOS, VPN apps require the use of the Network Extension framework, which must be signed with special entitlements granted by Apple. That means we need to sign the app and go through the App Store submission process.

Others can still use the code, but they'll need their own Apple Developer account with the necessary entitlements to build and sign the app themselves. Direct web distribution isn’t allowed for iOS apps with VPN functionality.

VpnHood Client distribution will also be available free for anyone who just wants to use the app without the hassle of compiling it themselves.

**Comment by trudyhood** [comment]
VpnHood is now available on iOS!
VpnHood! CLIENT: http://www.vpnhood.com/vpnhood-client/download/  
VpnHood! CONNECT: https://www.vpnhood.com/free-vpn/download/

Thank you for your patience
